2023-04-11 07:50:52,158 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-11 08:10:30,938 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-11 08:10:40,978 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-12 11:28:17,671 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-12 12:40:52,884 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-12 12:41:37,634 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-12 12:58:30,504 INFO: Microblog startup [in /Users/jocokiss/flask_project/application/__init__.py:47]
2023-04-12 12:58:31,808 ERROR: Exception on / [GET] [in /Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py:1744]
Traceback (most recent call last):
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/application/routes.py", line 31, in index
    return render_template('index.html', title='Home', posts=posts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/templating.py", line 147, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/templating.py", line 130, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/jocokiss/flask_project/application/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/jocokiss/flask_project/application/templates/base.html", line 31, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/application/templates/index.html", line 6, in block 'content'
    {{ form.hidden_tag() }}
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'form' is undefined
2023-04-12 12:58:37,970 ERROR: Exception on /index [GET] [in /Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py:1744]
Traceback (most recent call last):
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/application/routes.py", line 31, in index
    return render_template('index.html', title='Home', posts=posts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/templating.py", line 147, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/templating.py", line 130, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/jocokiss/flask_project/application/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/jocokiss/flask_project/application/templates/base.html", line 31, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/application/templates/index.html", line 6, in block 'content'
    {{ form.hidden_tag() }}
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'form' is undefined
2023-04-12 12:58:41,714 ERROR: Exception on /index [GET] [in /Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py:1744]
Traceback (most recent call last):
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/application/routes.py", line 31, in index
    return render_template('index.html', title='Home', posts=posts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/templating.py", line 147, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/flask/templating.py", line 130, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/jocokiss/flask_project/application/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/jocokiss/flask_project/application/templates/base.html", line 31, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jocokiss/flask_project/application/templates/index.html", line 6, in block 'content'
    {{ form.hidden_tag() }}
  File "/Users/jocokiss/flask_project/venv/lib/python3.11/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'form' is undefined
